src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScans.kt

```diff
@@ -1,209 +1,133 @@
     }
 
-    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
-
-    override fun chapterListRequest(manga: SManga): Request {
-        val url = "$apiUrl${manga.url.replace("comics", "chapters")}"
-        return GET(url, headers)
+    override fun searchMangaFromElement(element: Element): SManga = mangaFromElement(element).apply {
+        title = element.selectFirst("h1")?.text().toString()
     }
 
-    override fun chapterListParse(response: Response): List<SChapter> {
-        val chapters = response.parseAs<List<ApiChapter>>()
-
-        val mangaUrl = response.request.url.toString()
-            .substringAfter(apiUrl)
-            .replace("chapters", "comics")
-
-        return chapters.map { it.toSChapter(mangaUrl) }
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        val response = client.newCall(searchMangaRequest(page, query, filters)).execute()
+        val mangaList = response.asJsoup().select(searchMangaSelector())
+            .map { searchMangaFromElement(it) }
+            .filter { it.title.lowercase().contains(query.lowercase()) }
+        return Observable.just(MangasPage(mangaList, false))
     }
 
-    override fun pageListParse(response: Response): List<Page> {
-        val document = response.asJsoup()
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        author = document.selectFirst("span:contains(Author) + span")?.text()
 
-        val chapterPages = document.select("#__NEXT_DATA__").html()
-            .parseAs<NextData<ApiChapterPages>>()
-            .props.pageProps.chapter.pages
+        val infoBlock = document.selectFirst("section div div")
 
-        val pages = chapterPages.parseAs<List<String>>()
+        if (infoBlock != null) {
+            val infoBlockChildren = infoBlock.children()
 
-        val cdnUrl = updatedCdnUrl(document)
+            infoBlockChildren[0].selectFirst("h1")?.text()?.let { title = it }
+            description = infoBlockChildren[2].text()
 
-        return pages.mapIndexed { idx, image ->
-            Page(idx, "", "$cdnUrl$image")
-        }
-    }
-
-    private fun updatedCdnUrl(document: Document): String {
-        val cdnUrlFromPage = document.selectFirst("main div.maxWidth img")
-            ?.attr("src")
-            ?.substringBefore("?")
-            ?.let { "$it?fileId=" }
-
-        return preferences.getCdnUrl()
-            .let {
-                if (it != cdnUrlFromPage && cdnUrlFromPage != null) {
-                    preferences.putCdnUrl(cdnUrlFromPage)
-                    cdnUrlFromPage
-                } else {
-                    it
+            with(infoBlockChildren[1].select("span")) {
+                status = when (this.removeAt(0)?.text()?.lowercase()) {
+                    "ongoing" -> SManga.ONGOING
+                    else -> SManga.UNKNOWN
                 }
+                genre = this.joinToString(", ") { it.text() }
```

Comment:
`, ` is the default and can be omitted.
```suggestion
                genre = this.joinToString { it.text() }
```